src/definitions/modules/calendar.js

```diff
@@ -1342,16 +1342,20 @@ $.fn.calendar.settings = {
       if (!text) {
         return null;
       }
-      text = ('' + text).trim().toLowerCase();
+      text = String(text).trim();
       if (text.length === 0) {
         return null;
       }
+      if(text.match(/^[0-9]{4}[\/\-\.][0-9]{2}[\/\-\.][0-9]{2}$/)){
+        text += ' 00:00:00';
+      }
       // Reverse date and month in some cases
-      text = settings.monthFirst ? text : text.replace(/[\/\-\.]/g,'/').replace(/([0-9]+)\/([0-9]+)/,'$2/$1');
+      text = settings.monthFirst || !text.match(/^[0-9]{2}[\/\-\.]/) ? text : text.replace(/[\/\-\.]/g,'/').replace(/([0-9]+)\/([0-9]+)/,'$2/$1');
       var textDate = new Date(text);
       if(!isNaN(textDate.getDate())) {
         return textDate;
       }
+      text = text.toLowerCase();
```

Comment:
I am asking because I am not familiwar with calendar module.
What is a reason that  `toLowerCase()` is required?

Reply:
In further parsing, calendar also checks for month names and 12h clock definitions (am/pm) against existing array strings. To support sensitive spelling, the text is set to lowercase before.

Reply:
Ah, I see!